From 165fe26996ca76480878ae79e377655e12268c6a Mon Sep 17 00:00:00 2001
From: Kevin Kofler <kevin.kofler@chello.at>
Date: Sun, 16 Oct 2016 18:44:22 +0200
Subject: [PATCH] [grubcfg] Fix mismatched quoting and escaping.

If we use .replace("'", "'\\''") for escaping, we also need to use
single quotes, not double quotes.

This was broken by the port from '%' to format, which also randomly
changed quoting characters for no good reason. Changing the outer ones
does not matter, but \" or ' within a string is not the same thing!
---
 src/modules/grubcfg/main.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/grubcfg/main.py b/src/modules/grubcfg/main.py
index 4e05a0d..1cc1f78 100644
--- a/src/modules/grubcfg/main.py
+++ b/src/modules/grubcfg/main.py
@@ -68,7 +68,7 @@ def modify_grub_default(partitions, root_mount_point, distributor):
     if swap_uuid:
         kernel_params.append("resume=UUID={!s}".format(swap_uuid))
 
-    distributor_line = "GRUB_DISTRIBUTOR=\"{!s}\"".format(distributor_replace)
+    distributor_line = "GRUB_DISTRIBUTOR='{!s}'".format(distributor_replace)
 
     if not os.path.exists(default_dir):
         os.mkdir(default_dir)
@@ -126,7 +126,7 @@ def modify_grub_default(partitions, root_mount_point, distributor):
                 else:
                     escaped_value = str(value).replace("'", "'\\''")
 
-                lines.append("{!s}=\"{!s}\"".format(key, escaped_value))
+                lines.append("{!s}='{!s}'".format(key, escaped_value))
 
     if not have_kernel_cmd:
         kernel_cmd = "GRUB_CMDLINE_LINUX_DEFAULT=\"{!s}\"".format(" ".join(kernel_params))
-- 
2.1.0

